File: src/main/java/com/sica/backend/repository/StudentRepository.java
package com.sica.backend.repository;

import com.sica.backend.entity.Student;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface StudentRepository extends JpaRepository<Student, String> {
    Optional<Student> findByEnrollmentNumber(String enrollmentNumber);
    Boolean existsByEnrollmentNumber(String enrollmentNumber);
}

File: src/main/kotlin/com/sica/backend/repository/StudentRepository.kt
package com.sica.backend.repository

import com.sica.backend.entity.Student
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface StudentRepository : JpaRepository<Student, String> {
    fun findByEnrollmentNumber(enrollmentNumber: String): Optional<Student>
    fun existsByEnrollmentNumber(enrollmentNumber: String): Boolean
}
